frontend/src/components/Pages/Todo/TodoList.jsx

```diff
@@ -1,23 +1,33 @@
 import "./todo.css";
 import React from "react";
 import TodoItem from "./TodoItem";
+import TodoItemGroup from "./TodoItemGroup";
 
-export function renderItems(items, ifEmpty = undefined) {
+export function renderItems(items, { ifEmpty, onItemToggled }) {
   var index = 0;
-  if (items.length !== 0) {
-    return items.map((todoItem) => (
-      <TodoItem key={index++} itemData={todoItem} />
-    ));
-  }
-  return ifEmpty && <li>{ifEmpty}</li>;
+  return (
+    <ul>
+      {items.length !== 0 ? (
+        items.map((todoItem) => {
```

Comment:
The map() method takes an index as a param. I suggest we use that for the keys, instead of `index++`

Reply:
`items.length !== 0` can also be changed to `items.length` since 0 is falsy.

Reply:
Good points! I didn't even realise `.map((element, index) => ...)` was a thing ❤️

Reply:
Yeah, the `.map()` method actually has 3 parameters, `element`, `index`, and `array`, for instance a way to find out if there are duplicates in an array:
```ts
anArray = [1, 2, 3, 4, 3, 6, 7]
duplicates = anArray.map((element, index, array) => array.indexOf(element) !== index)
```